peer/network.go

```diff
@@ -78,9 +78,10 @@ type Network interface {
 	// (length of response divided by request time), and with 0 if the response is invalid.
 	TrackBandwidth(nodeID ids.NodeID, bandwidth float64)
 
-	// NewAppProtocol reserves a protocol identifier and returns a corresponding
-	// client to send messages with
-	NewAppProtocol(protocol uint64, handler p2p.Handler, options ...p2p.ClientOption) (*p2p.Client, error)
+	// NewClient returns a client to send messages with for the given protocol
+	NewClient(protocol uint64, options ...p2p.ClientOption) (*p2p.Client, error)
+	// NewAppProtocol registers a server handler for an application protocol
+	AddHandler(protocol uint64, handler p2p.Handler) error
```

Comment:
q: could you comment on the difference between `AddHandler` and the various `SetGossip/Request/CrossChainRequest/Handler`?

Reply:
`AddHandler` is for the p2p sdk handlers. The existing signatures are for the legacy handlers. Eventually we will delete all the old stuff from this once migration is complete